flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -411,6 +411,23 @@ class TaskManager(
             log.debug(s"Cannot find task to fail for execution ${executionID})")
           }
 
+        // stops a task
+        case StopTask(executionID) =>
+          val task = runningTasks.get(executionID)
+          if (task != null) {
+            try {
+              task.stopExecution()
+              sender ! new TaskOperationResult(executionID, true)
+            } catch {
+              case t: Throwable =>
+                        sender ! new TaskOperationResult(executionID, false, t.getMessage())
```

Comment:
If only an `UnsupportedOperationException` can be thrown, then it is fine. To be honest, I don't know why the `TaskOperationResult` stores the exception as a string. I assume that it's some legacy artifact. I'm not a big fan of it. But you can use the `StringUtils.stringifyException` to also include the stack trace of the exception in the string.